### PR TITLE
Add support for freetype-gl

### PR DESCRIPTION
**Pull request by caseymcc**
* I've followed [this guide](https://hunter.readthedocs.io/en/latest/creating-new/create/cmake.html)
  step by step carefully. **Yes**

* I've tested this package remotely and have excluded all broken builds.
  Here is the links to the Travis/AppVeyor with status "All passed":

  * https://ci.appveyor.com/project/caseymcc/hunter-testing/builds/34793318
  * https://travis-ci.org/github/caseymcc/hunter-testing/builds/720109095

* I have submitted CI configs to https://github.com/cpp-pm/hunter-testing targeting `pkg.template` branch,
  see this merged pull request https://github.com/cpp-pm/hunter-testing/pull/74

Need a package repo for https://github.com/caseymcc/freetype-gl/tree/hunter and release.

### CODE REVIEW

**Comment by rbsheth** [comment]
@caseymcc https://github.com/cpp-pm/freetype-gl

**Comment by caseymcc** [comment]
PR created for package https://github.com/cpp-pm/freetype-gl/pull/1

**Comment by rbsheth** [comment]
@caseymcc https://github.com/cpp-pm/freetype-gl/releases/tag/v0.0.0-1a8c007-p0

**Comment by caseymcc** [comment]
@rbsheth update with the cpp-pm package release, pr for the test at https://github.com/cpp-pm/hunter-testing/pull/74

https://travis-ci.org/github/caseymcc/hunter-testing/builds/726012729
https://ci.appveyor.com/project/caseymcc/hunter-testing/builds/35136911

File: cmake/projects/freetype-gl/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+include(hunter_cacheable)
+include(hunter_cmake_args)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    freetype-gl
+    VERSION
+    "20.21.6-p0" #project has no releases so usinging date Y.M.D-patch
```
Comment (rbsheth) [inline cmake/projects/freetype-gl/hunter.cmake:13]: Version needs to match the release name. So `v0.0.0-1a8c007-p0`

File: cmake/projects/freetype/hunter.cmake [hunk 2]
```diff
@@ -8,6 +8,17 @@ include(hunter_cmake_args)
 include(hunter_download)
 include(hunter_pick_scheme)
 
+hunter_add_version(
+    PACKAGE_NAME
+    freetype
+    VERSION
+    "2.9.1-p4"
+    URL
+    "https://github.com/caseymcc/freetype/archive/v2.9.1-p4.tar.gz"
```
Comment (rbsheth) [inline cmake/projects/freetype/hunter.cmake:17]: PR needed for your freetype changes here: https://github.com/cpp-pm/freetype

**Review by rbsheth — CHANGES_REQUESTED** [review]

**Comment by caseymcc** [comment]
updated to use the github actions CI

**Comment by caseymcc** [comment]
@rbsheth What is missing for this PR?

File: cmake/projects/freetype-gl/ci/build.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Install Python package 'requests'
```
Comment (rbsheth) [inline cmake/projects/freetype-gl/ci/build.sh:1]: This should use the base script by default: https://hunter.readthedocs.io/en/latest/creating-new/test/ci.html#override-default-tests

Something like:
```
sudo apt-get install libgl1-mesa-dev

bash .github/workflows/ci/build.sh
```

File: cmake/projects/freetype-gl/hunter.cmake [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+include(hunter_cacheable)
+include(hunter_cmake_args)
+include(hunter_download)
+include(hunter_pick_scheme)
+
+hunter_add_version(
+    PACKAGE_NAME
+    freetype-gl
+    VERSION
+    "0.0.0-1a8c007-p0" #project has no releases so usinging date Y.M.D-patch
```
Comment (rbsheth) [inline cmake/projects/freetype-gl/hunter.cmake:13]: Nit: spelling

File: docs/spelling.txt [hunk 5]
```diff
@@ -43,7 +43,9 @@ decrypt
 decrypted
 doesn
 embeddable
+freetype
```
Comment (rbsheth) [inline docs/spelling.txt:46]: You can just add the keywords at the top of the package.rst file. Don't edit this one.

File: examples/freetype-gl/main.cpp [hunk 6]
```diff
@@ -0,0 +1,9 @@
+#include <iostream> // std::cout
```
Comment (rbsheth) [inline examples/freetype-gl/main.cpp:1]: You're not using iostream here?

File: examples/freetype-gl/main.cpp [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+#include "freetype-gl/freetype-gl.h"
+
+int main()
+{
+    texture_glyph_t *emptyGlph=texture_glyph_new();
```
Comment (rbsheth) [inline examples/freetype-gl/main.cpp:6]: Nit: spelling

File: docs/packages/pkg/freetype-gl.rst [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+.. index:: graphics ; freetype-gl
+
+.. _pkg.freetype-gl:
+
+freetype-gl
+===========
+
+-  `Official <https://github.com/rougier/freetype-gl>`__
+-  `Hunterized <https://github.com/hunter-packages/freetype-gl>`__
```
Comment (rbsheth) [inline docs/packages/pkg/freetype-gl.rst:13]: Not hunter-packages, cpp-pm

File: cmake/projects/freetype-gl/ci/matrix.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    { "example": "freetype-gl", "toolchain": "ninja-vs-15-2017-win64-cxx17",                    "os": "windows-2016", "python": "3.8", "script": "build.cmd" },
+    { "example": "freetype-gl", "toolchain": "nmake-vs-15-2017-win64-cxx17",                    "os": "windows-2016", "python": "3.8", "script": "build.cmd" },
+    { "example": "freetype-gl", "toolchain": "vs-16-2019-win64-cxx17",                          "os": "windows-2019", "python": "3.8", "script": "build.cmd" },
+    { "example": "freetype-gl", "toolchain": "vs-15-2017-win64-cxx17",                          "os": "windows-2016", "python": "3.8", "script": "build.cmd" },
+    { "example": "freetype-gl", "toolchain": "vs-14-2015-sdk-8-1",                              "os": "windows-2016", "python": "3.8", "script": "build.cmd" },
+    { "example": "freetype-gl", "toolchain": "mingw-cxx17",                                     "os": "windows-2016", "python": "3.8", "script": "build.cmd" },
+    { "example": "freetype-gl", "toolchain": "msys-cxx17",                                      "os": "windows-2016", "python": "3.8", "script": "build.cmd" }
+]
```
Comment (rbsheth) [inline cmake/projects/freetype-gl/ci/matrix.json:16]: Nit: newline at end

File: cmake/projects/freetype-gl/ci/build.sh [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+if [[ "$TOOLCHAIN" =~ "ios-nocodesign-13-2" ]]; then
+    export DEVELOPER_DIR="/Applications/Xcode_11.3.1.app/Contents/Developer"
+fi
+
+# Run build script
+if [[ "$BRANCH_NAME" == "master" ]]; then
+    python jenkins.py --upload
+else
+    python jenkins.py
+fi
```
Comment (rbsheth) [inline cmake/projects/freetype-gl/ci/build.sh:36]: Nit: newline at end

**Review by rbsheth — COMMENTED** [review]
@caseymcc couple small things need fixing. Then it will be merged!

**Review by rbsheth — APPROVED** [review]

**Comment by rahulshethsc** [comment]
Released: https://github.com/cpp-pm/hunter/releases/tag/v0.23.280

**Pull request merged**